rs/bitcoin/adapter/src/addressbook.rs

```diff
@@ -119,22 +122,31 @@ impl AddressBook {
     }
 
     /// This function takes the DNS seeds and creates a new queue of socket addresses to connect to
-    /// for the address discovery process.
-    fn build_seed_queue(&mut self) {
+    /// for the address discovery process. It is made async because `to_socket_addrs` is a blocking
+    /// operation that may involve DNS lookup.
+    async fn build_seed_queue(&mut self) -> Result<(), tokio::task::JoinError> {
         let mut rng = StdRng::from_entropy();
         let dns_seeds = self
             .dns_seeds
             .iter()
-            .map(|seed| format_addr(seed, self.port));
-        let mut addresses = dns_seeds
-            .flat_map(|seed| {
-                seed.to_socket_addrs().map_or(vec![], |v| {
-                    v.filter(|addr| !self.ipv6_only || addr.is_ipv6()).collect()
+            .map(|seed| format_addr(seed, self.port))
+            .collect::<Vec<_>>();
+        let ipv6_only = self.ipv6_only;
+        let tasks = dns_seeds
```

Comment:
NIT: It is probably not necessary that you collect the `dns_seeds` and then call `into_iter()` at them again. Probably, you can generate the tasks directly.

Reply:
Good point. Done!